kernel/examples/table-reader/src/main/java/io/delta/kernel/examples/MultiThreadedTableReader.java

```diff
@@ -226,11 +231,27 @@ private Runnable workConsumer(int workerId) {
                     if (work == ScanFile.POISON_PILL) {
                         return; // exit as there are no more work units
                     }
-                    try (CloseableIterator<FilteredColumnarBatch> dataIter = Scan.readData(
-                        tableClient,
-                        work.getScanRow(tableClient),
-                        Utils.singletonCloseableIterator(work.getScanFileRow(tableClient)),
-                        Optional.empty())) {
+                    Row scanState = work.getScanRow(tableClient);
+                    Row scanFile = work.getScanFileRow(tableClient);
+                    FileStatus fileStatus =
+                        InternalScanFileUtils.getAddFileStatus(scanFile);
+                    StructType physicalReadSchema =
+                        ScanStateRow.getPhysicalDataReadSchema(tableClient, scanState);
+
+                    try (
+                        CloseableIterator<ColumnarBatch> physicalDataIter =
+                            tableClient.getParquetHandler().readParquetFiles(
+                                singletonCloseableIterator(fileStatus),
+                                physicalReadSchema,
+                                Optional.empty()
+                            );
+                        CloseableIterator<FilteredColumnarBatch> dataIter =
+                            Scan.transformPhysicalData(
+                                tableClient,
+                                scanState,
+                                scanFile,
+                                physicalDataIter)
```

Comment:
It is the responsibility of the API that gets the iterator to close it. The first iterator `physicalDataIter` needn't be closed here. It could cause issues, but in this case it is ok given that the second iterator `dataIter` gets consumed within the `try` block.

I moved the `physicalDataIter` out of the `try..with..resources` block to simply reading the code. Now it should only be closed once as part of the API: `Scan.transformPhysicalData`.